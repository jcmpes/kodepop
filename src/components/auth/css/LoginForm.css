.login-form {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.form-title {
    position: relative;
    bottom: 20px;
}

.login-form-field {
    height: 3rem;
    width: 250px;
    padding: .5rem 1rem;
    margin-bottom: 1.5rem;
    font-size: 1.4rem;
    font-family: 'Helvetica neue', Helvetica, Arial, sans-serif;
    font-weight: 300;
    border: none;
    background-color: rgb(250, 250, 250);

}