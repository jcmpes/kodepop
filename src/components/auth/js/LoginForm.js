import React from 'react';
import PropTypes from 'prop-types';
import Button from '../../shared/Button';
import FormField from './FormField';
import { Spinner } from '../../shared';

import '../css/LoginForm.css'

function LoginForm({ onSubmit, loading, credentials, setCredentials, remember, setRemember }) {

    const handleInputChange = e => {
        setCredentials(oldCredentials => {
            return {
            ...oldCredentials,
            [e.target.name]: e.target.value
            }
        })
    }

    const handleCheckboxChange = e => {
        setRemember(e.target.checked);
    }

    const handleSubmit = e => {
        e.preventDefault();
        onSubmit(credentials, remember);
    }

    return (
        <React.Fragment>
            <h1 className="form-title">
                {loading ? 'Logging in...' : 'Log In' }
            </h1>
            <div>
                {loading ? <Spinner /> : 
                    <form className="login-form" onSubmit={handleSubmit}>
                        <FormField 
                            name="email"
                            placeholder="Email"
                            type="text"
                            className="login-form-field"
                            value={credentials.email}
                            onChange={handleInputChange}
                        />
                        <FormField 
                            name="password"
                            placeholder="Password"
                            type="password"
                            className="login-form-field"
                            value={credentials.password}
                            onChange={handleInputChange}
                        />
                        <FormField
                            name="remember"
                            id="remember"
                            label="Remember Me"
                            type="checkbox"
                            checked={remember}
                            onChange={handleCheckboxChange}
                        />
                        <Button
                            type="submit"
                            variant="highlight"
                            className="login-form-submit"
                            style={{ fontSize: '1.2rem' }}
                            disabled={!credentials.email || !credentials.password || loading}
                        >Submit</Button>
                    </form>
                }
            </div>
        </React.Fragment>
    )
}

LoginForm.propTypes = {
    onSubmit: PropTypes.func.isRequired,
    loading: PropTypes.bool.isRequired
}

export default LoginForm;