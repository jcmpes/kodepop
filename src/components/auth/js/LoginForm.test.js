import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from './LoginForm';

describe('LoginForm', () => {
  const props = {
    onSubmit: jest.fn(),
    loading: false,
    credentials: { email: 'JC', password: 'password' },
    setCredentials: jest.fn(),
    remember: false,
    setRemember: jest.fn()
  }

  const render = () => shallow(<LoginForm {...props} />);

  it('should render', () => {
    const wrapper = render();
    expect(wrapper.exists()).toBe(true);
  })

  it('should submit credentials and remember', () => {
    const wrapper = render();
    const emailField = wrapper.find('.login-form-field').at(0)
    emailField
      .props()
      .onChange({ target: { name: 'email', value: props.credentials.email } })
    const passwordField = wrapper.find('.login-form-field').at(1)
    passwordField
      .props()
      .onChange({ target: { name: 'password', value: props.credentials.password } })
    const rememberCheckbox = wrapper.find('#remember')
    rememberCheckbox
      .props()
      .onChange({ target: { name: 'remember', value: props.remember } })
    const form = wrapper.find('.login-form');
    form.simulate('submit', { preventDefault: () => {} });

    expect(props.onSubmit).toHaveBeenCalledWith(props.credentials, props.remember)
  })

})