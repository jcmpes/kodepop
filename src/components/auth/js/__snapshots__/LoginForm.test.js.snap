// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm snapshot testing snapshot testing 1`] = `
<Fragment>
  <h1
    className="form-title"
  >
    Log In
  </h1>
  <div>
    <form
      className="login-form"
      onSubmit={[Function]}
    >
      <FormField
        className="login-form-field"
        name="email"
        onChange={[Function]}
        placeholder="Email"
        type="text"
        value="JC"
      />
      <FormField
        className="login-form-field"
        name="password"
        onChange={[Function]}
        placeholder="Password"
        type="password"
        value="password"
      />
      <FormField
        checked={false}
        id="remember"
        label="Remember Me"
        name="remember"
        onChange={[Function]}
        type="checkbox"
      />
      <styled.button
        className="login-form-submit"
        disabled={false}
        style={
          Object {
            "fontSize": "1.2rem",
          }
        }
        type="submit"
        variant="highlight"
      >
        Submit
      </styled.button>
    </form>
  </div>
</Fragment>
`;
